### DISCUSSION TITLE
Senior Level: Question 6: When are segments tracked by Pimcore?  (bugged?)

### DISCUSSION THREAD

**Discussion by noolldave** [pimcore/pimcore#15968] [Q&A / Support] [answered]
Question 6: When are segments tracked by Pimcore?

- [ ] Always when opening Pimcore documents tagged with segments.
- [ ] When profiling rules with TrackSegment actions are configured.
- [ ] When SegmentTracker is used by the developer in the custom code.
- [ ] Always when opening the detail pages of data objects tagged with segments.

I've tried different combinations of answers and nothing is taken as correct.. 
is this now somehow buggy? 

According to the Pimcore Development Documentation:
- when a visitor opens a Pimcore document page with assigned segments, the segments are tracked automatically.
- when profiling rules with TrackSegment actions are configured, the segments are stored in the Targeting Storage of the current use.
- a developer can use SegmentTracker to track segments in custom code

if something is tagged with segments is tracked.. If SegmentTracker is in code, then I can track anything anytime when I want.. 

After being blocked 2 weeks because of this is quite frustrating.

**Comment by noolldave** [comment]
anybody any idea what could be wrong?

**Comment by twin-elements** [comment, marked as answer]
Hi
6. When are segments tracked by Pimcore?
 a. Always when opening Pimcore documents tagged with segments.
 b. When profiling rules with TrackSegment actions are configured.
 c. When SegmentTracker is used by the developer in the custom code.

**Reply by noolldave** [reply]
Awesome!!!
thank you very much.

However, I am pretty sure I tried those options, anyway it doesn't matter, the answer was accepted.
